controllers/authorino_controller.go

```diff
@@ -152,6 +151,11 @@ func (r *AuthorinoReconciler) Reconcile(ctx context.Context, req ctrl.Request) (
 				r.Client, statusNotReady(api.AuthorinoUpdatedReason, "Authorino Deployment resource updated"))
 			return ctrl.Result{RequeueAfter: time.Minute}, err
 		}
+
+		if !deploymentAvailable(existingDeployment) {
+			// Deployment not ready – return and requeue
+			return ctrl.Result{Requeue: true}, nil
```

Comment:
Have you considered watching owned deployments instead of requeueing when the status is not ready?

Reply:
TBH, it never occurred to me, but it sounds an interesting idea.

Reply:
Should we define another reconciler for the Deployment?

The reconciler of the Authorino CR would no longer update the status, I suppose. We'd probably need to trace back from the Deployment to the Authorino CR in the new reconciler. Then, check whether the Deployment is still valid. In case it is and the status of the Deployment is Ready, we update the status of the CR?

Is that the idea? Isn't it too much work? Or perhaps you had something else in mind.